src/store.c

```diff
@@ -1030,15 +1039,30 @@ void pthreads_store_reset(zval *object, HashPosition *position) {
 
 	if (pthreads_monitor_lock(threaded->monitor)) {
 		zend_hash_internal_pointer_reset_ex(threaded->store.props, position);
+		if (zend_hash_has_more_elements_ex(threaded->store.props, position) == FAILURE) { //empty
+			*position = HT_INVALID_IDX;
```

Comment:
This is duct tape and should probably be fixed better later.

Reply:
Yes, it's not a beauty but it is working for now. And who knows when they will change this in php again.